packages/database/src/utilities/runDatabaseFunctionInBatches.js

```diff
@@ -3,18 +3,21 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import flatten from 'lodash.flatten';
+
 // some part of knex or node-pg struggles with too many bindings, so we batch them in some places
-export const MAX_BINDINGS_PER_QUERY = 2500; // errors occurred at around 5000 bindings in testing
+// errors occurred at around 5000 bindings in initial testing, but more is needed - see issue #2197
+export const MAX_BINDINGS_PER_QUERY = 2500;
 
-export async function runDatabaseFunctionInBatches(arrayToBeBatched, databaseFunction) {
-  const result = [];
-  const batchSize = MAX_BINDINGS_PER_QUERY;
+export async function runDatabaseFunctionInBatches(
+  arrayToBeBatched,
+  databaseFunction,
+  batchSize = MAX_BINDINGS_PER_QUERY,
+) {
+  const batches = [];
   for (let i = 0; i < arrayToBeBatched.length; i += batchSize) {
-    const batch = arrayToBeBatched.slice(i, i + batchSize);
-    const resultForBatch = await databaseFunction(batch);
-    if (Array.isArray(resultForBatch)) {
-      result.push(...resultForBatch);
-    }
+    batches.push(arrayToBeBatched.slice(i, i + batchSize));
   }
-  return result;
+  const batchedResults = await Promise.all(batches.map(async b => databaseFunction(b)));
+  return flatten(batchedResults.filter(r => Array.isArray(r)));
```

Comment:
This will work for 1 nesting level, which I guess is the case here... If not, we'd have to specify a max nesting level

```suggestion
  return batchedResults.filter(r => Array.isArray(r)).flat();
```

Reply:
Oh nice! Forgot about node's improvements 🙌 

Yep just one level. Will change that over and remove the dependency